src/webui/webapplication.cpp

```diff
@@ -573,8 +573,12 @@ void WebApplication::sendFile(const Path &path)
     }
 
     QByteArray data = readResult.value();
-    const QMimeType mimeType = QMimeDatabase().mimeTypeForFileNameAndData(path.data(), data);
-    const bool isTranslatable = !m_isAltUIUsed && mimeType.inherits(u"text/plain"_s);
+    QString mimeType = m_mimeTypes.value(path.extension().toLower());
+
+    if (mimeType.isNull())
+        mimeType = QMimeDatabase().mimeTypeForFileNameAndData(path.data(), data).name();
+
+    const bool isTranslatable = !m_isAltUIUsed && mimeType.startsWith(u"text/"_s);
```

Comment:
I doubt it is acceptable change. `mimeType.inherits(u"text/plain"_s)` has a much larger reach.


As I said earlier I won't approve fighting with mime database usage as a goal. I would keep using `QMimeType` instead of type names.

Reply:
Well, there is no other way around it.
If you disapprove, close #21363 as wontfix.

Reply:
I don't see how this could be possible. As far as I understand,

> keep using `QMimeType` instead of type names.

is mutually exclusive with

> hardcode some fileextension-to-mimetype mapping

because `QMimeType` requires the database to work.

Reply:
> I don't see how this could be possible. As far as I understand,
> 
> > keep using `QMimeType` instead of type names.
> 
> is mutually exclusive with
> 
> > hardcode some fileextension-to-mimetype mapping
> 
> because `QMimeType` requires the database to work.

As I mentioned, I don't consider abandoning `QMimeDatabase` as a goal. It is enough to prevent the use of incorrect file type associations. To do this, we need to get associations for important files (such as html, js, css) not from the system mime database (where they can be corrupted), but hard-code them instead. We need `QMimeDatabase::mimeTypeForName()` to achieve it which I believe isn't affected by the issue we want to avoid.

Reply:
> We need `QMimeDatabase::mimeTypeForName()` to achieve it which I believe isn't affected by the issue we want to avoid.

It is affected. Everything in `QMimeDatabase` is affected. Because all types can be overridden in a user local database.

Reply:
And the whole thing seems hardcoded. I don't see a way to forbid it loading external databases.

https://github.com/qt/qtbase/blob/8b3865e15120d1f85590cc6e1a5ca9fa41697705/src/corelib/mimetypes/qmimedatabase.cpp#L72-L171

Reply:
So, there is a dead end here? Should I close the PR?

Reply:
I expressed my opinion. We can wait for someone else.

Reply:
I guess nobody else wants to participate.